conans/test/unittests/model/build_info/components_test.py

```diff
@@ -50,7 +49,7 @@ def test_deps_cpp_info_libs(self):
         self.assertListEqual(["libc", "libd"], deps_cpp_info["dep2"].libs)
         self.assertListEqual(["libdep3"], deps_cpp_info["dep3"].libs)
         self.assertListEqual(["liba", "libb", "libc", "libd", "libdep3"],
-                             deps_cpp_info.libs)
+                             list(deps_cpp_info.libs))
```

Comment:
I don't get why this change is needed. Isn't `deps_cpp_info.libs` already a list?

Reply:
`libs` attribute from any other object but `cpp_info` itself is a read-only field.

Right now it is a `list`, agree, but that is the worst choice for a read-only field, implementation could change to `tuple` or even a `generator` and the test shouldn't fail because of that _implementation detail_. That's the reason why I'm calling `list`, to expand a possible generator or change a tuple into a list.

Reply:
Actually, I think this conversion to ``list()`` in the test, that allows changing the implementation of ``.libs``, ``.include_paths`` and other attributes to a different thing other than a list, is a risky thing. Because it is explicitly documented that these things are lists: https://docs.conan.io/en/latest/reference/conanfile/attributes.html#cpp-info

We could have the tests green, but break users without knowing. I think the ``list()`` conversions need to be removed, to guarantee in tests that these attribute return lists.